File: LeetCode/2215. 找出两数组的不同.go
func findDifference(nums1 []int, nums2 []int) [][]int {
    n1,n2 := len(nums1),len(nums2)
    m1 := make(map[int]bool)
    m2 := make(map[int]bool)
    for i:=0;i<n1;i++ {
        m1[nums1[i]] = true
    }
    for i:=0;i<n2;i++ {
        m2[nums2[i]] = true
    }
    res := make([][]int, 2)
    for k,_ := range m1 {
        if !m2[k] {
           res[0] = append(res[0], k) 
        }
    }
     for k,_ := range m2 {
        if !m1[k] {
           res[1] = append(res[1], k) 
        }
    }
    return res
}

File: LeetCode/2215. 找出两数组的不同.cpp
class Solution {
public:
vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> m1, m2;
        for (int i : nums1)
        {
            m1.insert(i);
        }
        for (int i : nums2)
        {
            m2.insert(i);
        }
        vector<vector<int>> answer(2);
        for (int i : m1)
        {
            if (!m2.count(i)) {
                answer[0].push_back(i);
            }
        }
        for (int i : m2)
        {
            if (!m1.count(i)) {
                answer[1].push_back(i);
            }
        }
        return answer;
    }
};